fdbserver/BlobManager.actor.cpp

```diff
@@ -3498,6 +3498,31 @@ ACTOR Future<Void> loadBlobGranuleMergeBoundaries(Reference<BlobManagerData> bmD
 	return Void();
 }
 
+// Update blob manager epoc for restore. The blob manager should have larger epoch after restore
+// so that it could generate new manifest files with larger epoch number. Blob manifest files
+// with largest epoch number is treated as the newest.
+ACTOR Future<Void> updateEpoch(Reference<BlobManagerData> bmData, int64_t newEpoch) {
+	// Skip update if the expected epoc is less than current epoch
+	if (newEpoch <= bmData->epoch) {
+		return Void();
```

Comment:
I think we need to do this as part of the transaction. This blob manager could be running with an old epoch and overwrite the database's epoch which has already increased to a higher value.
If you just do a get on blobManagerEpoch key before the set and do the same check as part of the transaction it should work

Reply:
ok. fixed. thank you!